models/crypto.js

```diff
@@ -0,0 +1,70 @@
+const firestore = require('../utils/firestore')
+const cryptoProductsCollection = firestore.collection('crypto-products')
+
+/**
+ * Fetches the data of crypto product
+ * @return {Promise<product|Object>}
+ */
+
+const fetchProducts = async () => {
+  try {
+    const snapshot = await cryptoProductsCollection.get()
+
+    const productsData = {}
+
+    snapshot.forEach((doc) => {
+      const data = doc.data()
+      productsData[data.id] = { ...data }
+    })
+    return productsData
+  } catch (err) {
+    logger.error('Error retrieving products data', err)
+    throw err
+  }
+}
+
+/**
+ * Fetches the data of crypto product
+ * @param productData {object} - product details that is to be put in DB
+ * @return {Promise<product|object>}
+ */
+const addProduct = async (productData) => {
+  try {
+    const { id } = productData
+    const product = await cryptoProductsCollection.doc(id).get()
+    if (!product.exists) {
+      await cryptoProductsCollection.doc(id).set(productData)
+      return productData
+    } else {
+      return undefined
```

Comment:
By default it will return undefined, we can return proper message or error

Reply:
Yes but that will give a linting error, This is DAO, is returning an error message necessary hear